rover_video/src/media_server.cpp

```diff
@@ -0,0 +1,117 @@
+#include "screenshot_server.hpp"
+
+int main(int argc, char* argv[])
+{
+    rclcpp::init(argc, argv);
+
+    RCLCPP_INFO(rclcpp::get_logger("media_server"), "Node initialized.");
+
+    try
+    {
+        rclcpp::spin(std::make_shared<CameraNode>());
+    }
+    catch (const std::exception& e)
+    {
+        RCLCPP_FATAL(rclcpp::get_logger("Dead Node"), "Killing node on exception: %s", e.what());
+    }
+
+    rclcpp::shutdown();
+    return 0;
+}
+
+CameraNode::CameraNode():
+    Node("media_server")
+{
+    _srv_control = this->create_service<rover_msgs::srv::CameraControl>(
+        "/rover/video/media_server",
+        [this](const std::shared_ptr<rover_msgs::srv::CameraControl::Request> request,
+               std::shared_ptr<rover_msgs::srv::CameraControl::Response> response)
+        {
+            this->controlIPCam(*request, *response);
+        });
+
+    _msg_position
+        = this->create_subscription<rover_msgs::msg::GpsPosition>("/rover/gps/position",
+                                                                  1,  // What to put as QoS ?
+                                                                  [this](const rover_msgs::msg::GpsPosition& gps_message)
+                                                                  {
+                                                                      this->callbackPosition(gps_message);
+                                                                  });
+}
+
+void CameraNode::controlIPCam(const rover_msgs::srv::CameraControl::Request& request,
+                              rover_msgs::srv::CameraControl::Response& response)
+{
+    RCLCPP_DEBUG(LOGGER, "Entering the controlIPCam function");
+
+    std::string folderPath;
+    std::string captureName;
+    std::string cameraURL = request.camera_url;
+    // Définir screenshot ou video
+    switch (request.command)
+    {
+        case rover_msgs::srv::CameraControl::Request::TAKE_PICTURE:
+            captureName = getFileName(request.capture_name, cameraURL, SCREENSHOT);
+            folderPath = getFolderPath(SCREENSHOT);
+            if (!createFolder(folderPath))
+            {
+                RCLCPP_ERROR(LOGGER, "Failed to create screenshots folder or it already exists.");
+                response.success = false;
+                response.status = "Failed to create screenshots folder or it already exists.";
+                break;
+            }
+            if (getScreenshot(folderPath, captureName, cameraURL))
+            {
+                response.success = true;
+                response.status = "Screenshot saved as " + folderPath + "/" + captureName;
+            }
+            else
+            {
+                response.success = false;
+                response.status = "Failed to take a screenshot.";
+            }
+            break;
```

Comment:
This should be a whole function. It will make the flow a bit clearer since there's multiple exit point per cases.


```suggestion
        case rover_msgs::srv::CameraControl::Request::TAKE_PICTURE:
            response_ = this->takeScreenShot(request_);
            break;
        case rover_msgs::srv::CameraControl::Request::START_RECORDING:
            response_ = this->startRecording(request_);
            break;
        etc...
```